extensions/devservices/postgresql/src/main/java/io/quarkus/devservices/postgresql/deployment/PostgresqlDevServicesProcessor.java

```diff
@@ -70,6 +71,7 @@ public QuarkusPostgreSQLContainer(Optional<String> imageName, OptionalInt fixedE
                     .asCompatibleSubstituteFor(DockerImageName.parse(PostgreSQLContainer.IMAGE)));
             this.fixedExposedPort = fixedExposedPort;
             this.useSharedNetwork = useSharedNetwork;
+            withNetwork(Network.SHARED);
```

Comment:
PostgreSQL (and I suspect others) are not joining the _default_ Docker shared network setup by TestContainers.

This, I suspect should be the case, since other Quarkus services join it.

Reply:
This need to be an extra licit user configuration.

Reply:
Hello @cescoffier IIUC you're proposing to rollback my change here and add a similar mechanism to `KafkaBuildTimeConfig` where the "user" of the PostgreSQL DevService can decide what Docker network to have the service join? Happy to do so, I just want to be clearer on your requirement. I'd also love to understand your reasoning why some DevServices join TestContainers `SHARED` network already and others do not; so I can avoid making similar mistakes in the future.

Reply:
> This need to be an extra licit user configuration.

+1

Reply:
IIRC, none of the dev services are added to the `SHARED` network by default. The reason this was introduces was to support the use case of having `@QuarkusIntegrationTest` + DevServices

Reply:
Some are (mainly around messaging) see [here](https://github.com/quarkusio/quarkus/blob/main/extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/DevServicesKafkaProcessor.java#L353) as an example. Anyway, no problem, I can add the User configuration instead once @cescoffier confirms my understanding.

Reply:
That said, adding them to the `SHARED` network by default should not cause any problems.

Reply:
@cescoffier @geoand I am thinking of adding a new property to `io.quarkus.datasource.runtime.DevServicesBuildTimeConfig` called `useTestContainerSharedNetwork`. This would be extracted in `io.quarkus.datasource.deployment.devservices.DevServicesDatasourceProcessor.startDevDb(..)` and passed to `devDbProvider.startDatabase(...)` along with the other properties; making necessary changes elsewhere to pass to the various DataSource container implementations. WDYT? A bit more extensive than me simply bolting something on the my PostgreSQL use-case; but at least it'd bring consistency to the different DataSources supported by DevServices. 

To be honest I'd like to see a refactor of Quarkus' `useSharedNetwork` flag - since its interpretation is not analogous with `TestContainers` shared network and it really drives how services are exposed outside the Docker network. Anyhow.. that's not a discussion for today. That said it is _internal_ and I only found it confusing as I was trying to work out my issue.

Reply:
> WDYT?

Sounds good to me

Reply:
OK, I've added the change as a new commit to simplify (or so I hope) review.

Some containers that jump on `TestContainer`'s shared network remain:-
```
./extensions/smallrye-reactive-messaging-amqp/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/amqp/deployment/AmqpDevServicesProcessor.java:294:            withNetwork(Network.SHARED);
./extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/DevServicesKafkaProcessor.java:353:            withNetwork(Network.SHARED);
./extensions/vault/deployment/src/main/java/io/quarkus/vault/deployment/DevServicesVaultProcessor.java:148:        vaultContainer.withNetwork(Network.SHARED);
./extensions/vault/deployment/src/main/java/io/quarkus/vault/deployment/DevServicesVaultProcessor.java:209:            withNetwork(Network.SHARED);
./extensions/apicurio-registry-avro/deployment/src/main/java/io/quarkus/apicurio/registry/avro/DevServicesApicurioRegistryProcessor.java:267:            withNetwork(Network.SHARED);
./extensions/smallrye-reactive-messaging-rabbitmq/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/rabbitmq/deployment/RabbitMQDevServicesProcessor.java:374:            withNetwork(Network.SHARED);
```
Those I've "fixed":-
```
./extensions/devservices/mssql/src/main/java/io/quarkus/devservices/mssql/deployment/MSSQLDevServicesProcessor.java:78:                withNetwork(Network.SHARED);
./extensions/devservices/oracle/src/main/java/io/quarkus/devservices/oracle/deployment/OracleDevServicesProcessor.java:82:                withNetwork(Network.SHARED);
./extensions/devservices/mariadb/src/main/java/io/quarkus/devservices/mariadb/deployment/MariaDBDevServicesProcessor.java:77:                withNetwork(Network.SHARED);
./extensions/devservices/db2/src/main/java/io/quarkus/devservices/db2/deployment/DB2DevServicesProcessor.java:79:                withNetwork(Network.SHARED);
./extensions/devservices/postgresql/src/main/java/io/quarkus/devservices/postgresql/deployment/PostgresqlDevServicesProcessor.java:77:                withNetwork(Network.SHARED);
./extensions/devservices/mysql/src/main/java/io/quarkus/devservices/mysql/deployment/MySQLDevServicesProcessor.java:76:                withNetwork(Network.SHARED);
```
Some integration tests set it up too, but they're of less (read as "no") interest.
```
./integration-tests/kafka-oauth-keycloak/src/test/java/io/quarkus/it/kafka/containers/KafkaContainer.java:60:        withNetwork(Network.SHARED);
./integration-tests/kafka-oauth-keycloak/src/test/java/io/quarkus/it/kafka/containers/KeycloakContainer.java:24:        withNetwork(Network.SHARED);
./integration-tests/kafka-streams/src/test/java/io/quarkus/it/kafka/streams/KafkaSSLTestResource.java:93:            super.withNetwork(Network.SHARED);
./integration-tests/kafka-ssl/src/test/java/io/quarkus/it/kafka/KafkaSSLTestResource.java:91:            super.withNetwork(Network.SHARED);
./integration-tests/kafka-sasl-elytron/src/test/java/io/quarkus/it/kafka/containers/KafkaContainer.java:63:        withNetwork(Network.SHARED);
./integration-tests/kafka-sasl-elytron/src/test/java/io/quarkus/it/kafka/containers/KerberosContainer.java:27:        withNetwork(Network.SHARED);
./integration-tests/kafka-sasl/src/test/java/io/quarkus/it/kafka/KafkaSASLTestResource.java:83:            super.withNetwork(Network.SHARED);
```